### PR TITLE
apps: add gatekeeper psps for kured

### PR DESCRIPTION
**Pull request by OlleLarsson**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/values/podsecuritypolicies/common/kured.gotmpl [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  kured:
+    kured:
+      podSelectorLabels:
+        app.kubernetes.io/instance: kured
+      allow:
+        hostNamespace: true
+        allowPrivilegeEscalation: true
+        privileged: true
+        runAsUser:
+          rule: RunAsAny
```
Comment (aarnq) [inline helmfile/values/podsecuritypolicies/common/kured.gotmpl:11]:
```suggestion
{{- if .Values.kured.enabled }}
constraints:
  kured:
    kured:
      podSelectorLabels:
        app.kubernetes.io/instance: kured
      allow:
        hostNamespace: true
        allowPrivilegeEscalation: true
        privileged: true
        runAsUser:
          rule: RunAsAny
{{- end }}
```
(If that is the correct key.)

File: helmfile/charts/gatekeeper/podsecuritypolicies/templates/custom/mutations/privilege-escalation.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{{- range $namespace, $services := .Values.constraints }}
+{{- range $name, $value := $services }}
+{{- if get $value.allow "allowPrivilegeEscalation" }}
+---
+apiVersion: mutations.gatekeeper.sh/v1
+kind: Assign
+metadata:
+  name: {{ printf "privilege-escalation-%s" $name | trunc 63 }}
```
Comment (aarnq) [inline helmfile/charts/gatekeeper/podsecuritypolicies/templates/custom/mutations/privilege-escalation.yaml:8]: The way I've done with the rest is:
```suggestion
  name: {{ printf "privilege-escalation-%s-container" $name | trunc 63 }}
```
And then the same with `ephemeral` and `init` so each are grouped together based on type and then on service.

Comment (OlleLarsson) [reply helmfile/charts/gatekeeper/podsecuritypolicies/templates/custom/mutations/privilege-escalation.yaml]: I have updated, PTAL

**Review by aarnq — APPROVED** [review]
LGTM

**Review by robinAwallace — APPROVED** [review]

**Pull request merged**